### PR TITLE
make libsql client a valid target for cloudflare workers

### PR DESCRIPTION
**Pull request by Horusiath**
This PR introduces new feature: `cloudflare-worker`, which enables Hrana protocol to be served over Cloudflare workers.

### CODE REVIEW

File: libsql/src/connection.rs [hunk 1]
```diff
@@ -6,7 +6,8 @@ use crate::statement::Statement;
 use crate::transaction::Transaction;
 use crate::{Result, TransactionBehavior};
 
-#[async_trait::async_trait]
+#[cfg_attr(feature = "cloudflare-worker", async_trait::async_trait(?Send))]
+#[cfg_attr(not(feature = "cloudflare-worker"), async_trait::async_trait)]
```
Comment (Horusiath) [inline libsql/src/connection.rs:10]: Any advice on how to make it cleaner? @LucioFranco ?

File: libsql/src/local/impls.rs [hunk 2]
```diff
@@ -15,7 +15,8 @@ pub(crate) struct LibsqlConnection {
     pub(crate) conn: super::Connection,
 }
 
-#[async_trait::async_trait]
+#[cfg_attr(feature = "hrana-cf", async_trait::async_trait(?Send))]
+#[cfg_attr(not(feature = "hrana-cf"), async_trait::async_trait)]
```
Comment (LucioFranco) [inline libsql/src/local/impls.rs:19]: I don't think this is the right approach we should take since we need these features to be additive.

**Review by LucioFranco — CHANGES_REQUESTED** [review]
No real request changes yet because I don't know exactly what path we should take.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 /target
 .vscode/
+.idea/
```
[the remark this answers is not in the document]
Comment (Horusiath) [reply .gitignore:3]: Same could be said about `.vscode/` path ;)

**Comment by Horusiath** [comment]
1.

**Pull request closed without merging**